src/plugin_wrapper.rs

```diff
@@ -44,23 +46,32 @@ impl SimElement for PluginWrapper {
 impl PluginWrapper {
     pub fn new(plugin_path : &str, id : u32) -> PluginWrapper {
         let dyn_lib = libloading::Library::new(plugin_path).expect("load library");
+        
```

Comment:
Why do we need 2 unsafe blocks here instead of the 1 unsafe block we had earlier?

Reply:
There are now two external functions we are importing;  a new() function that initializes the hash table, and the execute() function that executes the filter functionality.

Reply:
Can we add a comment to this effect then? And maybe we should call the 2 functions filter_function and initializer_function to make it clear that they are both functions. Right now, the term filter doesn't make it clear it's a function and the term loaded_function is too generic.